### PR TITLE
fix: wrong number of products for lists + analytics events

### LINKED ISSUE #4647: Incorrect number of products in a list
### What
After a product added to a list, the "number of products" counter is invalid

### Steps to reproduce the behavior
(optional, delete if empty)
1. Open the Lists tab
2. Go back to the homescreen and search for a product
3. Select a product
4. Click on "Add to list"
5. Go back to the Lists tab
6. The number of elements in the list is incorrect

### Screenshot/Mockup/Before-After
https://user-images.githubusercontent.com/246838/162414168-a23cabbd-08d0-4fa7-8faf-731fc9cce8f1.mp4

### PR DESCRIPTION
**Pull request by g123k**
Fix for #4647 and will partially implement #4640

ProductsListsDao was never updated after a product addition/removal
+ analytics events when a product is added / removed
+ analytics event when a list is shared

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/analytics/AnalyticsEvent.kt [hunk 1]
```diff
@@ -26,7 +26,13 @@ sealed class AnalyticsEvent(val category: String, val action: String, val name:
 
     object RobotoffLoggedInAfterPrompt : AnalyticsEvent("user-account", "logged-in-after-prompt", "robotoff", null)
 
-    object ShoppingListCreated : AnalyticsEvent("shopping-lists", "created", null, null)
+    data class ShoppingListCreated(val listName: String?) : AnalyticsEvent("shopping-lists", "created", listName, null)
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/analytics/AnalyticsEvent.kt]: Do we really want to know about the product list name? Isn't it something a little bit privacy-concerning?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/summary/SummaryProductFragment.kt [hunk 2]
```diff
@@ -884,6 +884,11 @@ class SummaryProductFragment : BaseFragment(), ISummaryProductPresenter.View {
                 it.imageUrl = product.getImageSmallUrl(localeManager.getLanguage())
             }
             daoSession.listedProductDao.insertOrReplace(product)
+            daoSession.productListsDao.update(list.apply {
+                products.add(product)
+                numOfProducts++ }
+            )
+            matomoAnalytics.trackEvent(AnalyticsEvent.ShoppingListProductAdded(product.barcode))
             dialog.dismiss()
             onRefresh()
```
Comment (VaiTon) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/summary/SummaryProductFragment.kt:890]: Can we extract this to a fun?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/analytics/AnalyticsEvent.kt (hunk 1 above)
Comment (g123k) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/analytics/AnalyticsEvent.kt]: My idea behind thisn was to differentiate what kind of list the user create.
But clearly I can remove it if neccessary. What do you think @teolemon?

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/summary/SummaryProductFragment.kt (hunk 2 above)
Comment (g123k) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/features/product/view/summary/SummaryProductFragment.kt:890]: No pb

**Comment by g123k** [comment]
That should be ok now @VaiTon

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/analytics/AnalyticsEvent.kt (hunk 1 above)
Comment (teolemon) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/analytics/AnalyticsEvent.kt]: probably overkill. We're more focused on volumetry.

**Comment by g123k** [comment]
Following @teolemon answer, no list name is sent with the list creation analytics event

**Review by VaiTon — APPROVED** [review]

**Pull request merged**